internal/worker/worker.go

```diff
@@ -64,43 +74,63 @@ func (m *MultiWriterWithWebsocket) Write(p []byte) (n int, err error) {
 	if err := json.Unmarshal(p, &logMsg); err == nil {
 		// If it's a JSON log message, convert to plain text
 		message := fmt.Sprintf("[%s] %s\n", strings.ToUpper(logMsg.Level), strings.TrimSpace(logMsg.Message))
+		if m.prefix != "" {
+			message = fmt.Sprintf("[%s] %s", m.prefix, message)
+		}
 
 		// Write to the regular writer
 		n, err = m.writer.Write([]byte(message))
 		if err != nil {
 			return n, err
 		}
 
-		// Send to the websocket
-		if err := websocket.Message.Send(m.ws, message); err != nil {
-			// We've already written to the regular writer, so we can't return 0
-			// Instead, we'll return the number of bytes written to the regular writer
-			return n, err
+		// Send to the websocket with retry if available
+		if m.ws != nil {
```

Comment:
we only send over the wire, if we have a connection established. Initially the worker doesn't yet have a ws connection. But we still use the same writer everywhere.